central/sensor/service/connection/connection_test.go

```diff
@@ -442,6 +442,84 @@ func (s *testSuite) TestIssueLocalScannerCerts() {
 	}
 }
 
+func (s *testSuite) TestIssueSecuredClusterCerts() {
+	namespace, clusterID, requestID := "namespace", "clusterID", "requestID"
+	testCases := map[string]struct {
+		requestID  string
+		namespace  string
+		clusterID  string
+		shouldFail bool
+	}{
+		"no parameter missing": {requestID: requestID, namespace: namespace, clusterID: clusterID, shouldFail: false},
+		"requestID missing":    {requestID: "", namespace: namespace, clusterID: clusterID, shouldFail: true},
+		"namespace missing":    {requestID: requestID, namespace: "", clusterID: clusterID, shouldFail: true},
+		"clusterID missing":    {requestID: requestID, namespace: namespace, clusterID: "", shouldFail: true},
+	}
+	for tcName, tc := range testCases {
+		s.Run(tcName, func() {
+			sendC := make(chan *central.MsgToSensor)
+			sensorMockConn := &sensorConnection{
+				clusterID: tc.clusterID,
+				sendC:     sendC,
+				stopSig:   concurrency.NewErrorSignal(),
+				sensorHello: &central.SensorHello{
+					DeploymentIdentification: &storage.SensorDeploymentIdentification{
+						AppNamespace: tc.namespace,
+					},
+				},
+			}
+			ctx := context.Background()
+			ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+			defer cancel()
+			request := &central.MsgFromSensor{
+				Msg: &central.MsgFromSensor_IssueSecuredClusterCertsRequest{
+					IssueSecuredClusterCertsRequest: &central.IssueSecuredClusterCertsRequest{
+						RequestId: tc.requestID,
+					},
+				},
+			}
+
+			handleDoneErrSig := concurrency.NewErrorSignal()
+			go func() {
+				handleDoneErrSig.SignalWithError(sensorMockConn.handleMessage(ctx, request))
+			}()
+
+			select {
+			case msgToSensor := <-sendC:
+				response := msgToSensor.GetIssueSecuredClusterCertsResponse()
+				s.Equal(tc.requestID, response.GetRequestId())
+				if tc.shouldFail {
+					s.NotNil(response.GetError())
+				} else {
+					s.NotNil(response.GetCertificates())
+
+					certificates := response.GetCertificates()
+					s.NotNil(certificates.GetServiceCerts())
+					s.NotNil(certificates.GetCaPem())
+
+					serviceCertificates := certificates.GetServiceCerts()
+					expectedCertificates := 7
+					s.Equal(expectedCertificates, len(serviceCertificates))
+
+					for _, serviceCertificate := range serviceCertificates {
+						cert := serviceCertificate.GetCert()
+
+						s.NotNil(cert.GetCertPem())
```

Comment:
Nit: I'll throw in here a basic check that the cert can be loaded, that makes sure there is no corruption.

Reply:
Done